internal/services/ingest.go

```diff
@@ -374,6 +378,53 @@ func (m *ingestService) getLedgerTransactions(ctx context.Context, xdrLedgerClos
 	return transactions, nil
 }
 
+func (m *ingestService) ingestProcessedData(ctx context.Context, ledgerIndexer *indexer.Indexer) error {
+	dbTxErr := db.RunInTransaction(ctx, m.models.DB, nil, func(dbTx db.Transaction) error {
+		indexerBuffer := &ledgerIndexer.IndexerBuffer
+
+		// 1. Filter participants that are not in the watchlist.
+		// TODO: cache the existing accounts in memory and use the cache while processing the ledger.
+		existingAccounts, err := m.models.Account.GetExisting(ctx, dbTx, indexerBuffer.Participants.ToSlice())
+		if err != nil {
+			return fmt.Errorf("getting existing accounts: %w", err)
+		}
+
+		if len(existingAccounts) == 0 {
+			log.Ctx(ctx).Debug("🟡 no existing accounts found, skipping ingestion")
+			return nil
+		}
+
+		// 2. Identify which data should be ingested.
+		txHashesToInsert := set.NewSet[string]()
+		for _, participant := range existingAccounts {
+			if !indexerBuffer.Participants.Contains(participant) {
+				continue
+			}
+
+			// 2.1. Identify which transactions should be ingested.
+			participantTxHashes := indexerBuffer.GetParticipantTransactionHashes(participant)
+			for txHash := range participantTxHashes.Iterator().C {
+				txHashesToInsert.Add(txHash)
+			}
+
+			// 2.2. TODO: Identify which operations should be ingested.
```

Comment:
I am guessing this is the loop where we will also process the state changes using the filtered txns and ops. I do want to highlight that both the processors [take](https://github.com/stellar/wallet-backend/pull/195/files#diff-ddd0fefbb0615611beab0cd822e8d08b6ef11936cc9a26031d4a9f6295cff611R36) in `ingest.LedgerTransaction` so we do need to have that available in this FOR loop

Correct me if I am wrong but the flow looks like this - for each participant, we get the necessary txns, we pass the `ingest.LedgerTransaction` version of the txn to our processors, build the state changes and then insert all txns, ops, state changes together. And we make sure we dont process txns more than once by using the txHashesToInsert map